pypeit/tests/test_specutils.py

```diff
@@ -0,0 +1,136 @@
+    ofile = Path(tstutils.data_path('tmp.fits')).resolve()
+    spec.to_file(str(ofile), overwrite=True)
+
+    _spec = Spectrum1D.read(ofile)
+    assert np.array_equal(spec.flux, _spec.flux.data), 'Flux munged'
+    assert np.array_equal(spec.wave, _spec.spectral_axis.data), 'Wavelengths munged'
+
+    _spec = Spectrum1D.read(ofile, grid=True)
+    assert np.array_equal(spec.wave_grid_mid, _spec.spectral_axis.data), 'Wavelengths munged'
+
+    ofile.unlink()
+
+
+@specutils_required
+def test_spec1d_io():
+
+    ofile = Path(tstutils.data_path('tmp.fits')).resolve()
+
+    spec1 = specobj.SpecObj('MultiSlit', 'DET01', SLITID=0)
+    npix_spec = 100
+
+    # NOTE: WAVE, COUNTS, IVAR, and MASK must *all* exist to be able to read the
+    # file using specutils because of how the pypeit loader works (i.e., this
+    # isn't something intrinsic to specutils).
+    spec1['BOX_WAVE'] = np.arange(npix_spec).astype(float)
+    spec1['BOX_COUNTS'] = np.ones(npix_spec, dtype=float)
+    spec1['BOX_COUNTS_IVAR'] = np.ones(npix_spec, dtype=float)
+    spec1['BOX_MASK'] = np.ones(npix_spec, dtype=bool)
+    spec1['DETECTOR'] = tstutils.get_kastb_detector()
+
+    allspec = specobjs.SpecObjs([spec1])
+    header = fits.PrimaryHDU().header
+    header['TST'] = 'TEST'
+    allspec.write_to_fits(header, str(ofile), overwrite=True)
+
+    spec = SpectrumList.read(ofile)
+    assert len(spec) == 1, 'Should be 1 spectrum'
+    assert np.array_equal(spec[0].flux.data, spec1.BOX_COUNTS), 'Bad flux read'
+    assert spec[0].meta['name'] == 'SPAT-----SLIT0000-DET01', 'Name changed'
+    assert spec[0].meta['extract'] == 'BOX', 'Should have read spectrum as a boxcar extraction'
+    assert not spec[0].meta['fluxed'], 'Should have read spectrum as uncalibrated counts'
+
+    spec1['OPT_WAVE'] = spec1['BOX_WAVE'].copy()
+    spec1['OPT_COUNTS'] = spec1['BOX_COUNTS'] * 1.1
+    spec1['OPT_COUNTS_IVAR'] = spec1['BOX_COUNTS_IVAR'] * 1.1
+    spec1['OPT_MASK'] = spec1['BOX_MASK'].copy()
+
+    allspec.write_to_fits(header, str(ofile), overwrite=True)
+    spec = SpectrumList.read(ofile)
+    assert np.array_equal(spec[0].flux.data, spec1.OPT_COUNTS), 'Bad flux read'
+    assert spec[0].meta['extract'] == 'OPT', 'Should have read spectrum as an optimal extraction'
+
+    spec1['OPT_FLAM'] = spec1['OPT_COUNTS'] * 1.1
+    spec1['OPT_FLAM_IVAR'] = spec1['OPT_COUNTS_IVAR'] * 1.1
+
+    allspec.write_to_fits(header, str(ofile), overwrite=True)
+    spec = SpectrumList.read(ofile)
+    assert np.array_equal(spec[0].flux.data, spec1.OPT_FLAM), 'Bad flux read'
+    assert spec[0].meta['extract'] == 'OPT', 'Should have read spectrum as an optimal extraction'
+    assert spec[0].meta['fluxed'], 'Should have read the flux-calibrated data'
+
+    # Try reading the flux-calibrated boxcar extraction
+    spec = SpectrumList.read(ofile, extract='BOX', fluxed=True)
+    # TODO: Issue warning when requested data is not available?
```

Comment:
What would this look like?

Reply:
For example, in the `pypeit_spec1d_loader`, there's a line:
`_ext, _cal = sobj.best_ext_match(extract=extract, fluxed=fluxed)`

You could imagine doing:
```
if _ext != extract or _cal != fluxed:
    msgs.warn('Requested extraction/calibration type not available, returning ...')
```

I think the reason I didn't do this is that `extract` and `fluxed` have default values.  So this might inundate users with inane warnings.  Happy to be convinced otherwise, though.